fix(pkg/strvals): preserve leading zeros in vals

rbwsam opened the pull request:
When passing values with "helm install --set" values with leading zeros are
preserved and not parsed as ints.

Closes #2693

rbwsam committed:
fix(pkg/strvals): preserve leading zeros in vals

When passing values with "helm install --set" values with leading zeros are
preserved and not parsed as ints.

Closes #2693

bacongobbler commented on pkg/strvals/parser.go:
```diff
@@ -308,8 +308,11 @@ func typedVal(v []rune) interface{} {
-	if iv, err := strconv.ParseInt(val, 10, 64); err == nil {
-		return iv
+	// If this value does not start with zero, try parsing it to an int
+	if len(val) != 0 && val[0] != 48 {
```
out of curiosity/lack of understanding, what does 48 translate to in this situation?

rbwsam replied:
48 is the ASCII value for `0`. Thanks for pointing this out, this prompted me to learn more about runes and I found out that a better implementation would be to use a rune literal `'0'` instead which better communicates the intent of the code.

rbwsam committed:
fix(pkg/strvals): use rune literal instead of ASCII

When checking that a value begins with zero use a rune literal instead of the
ASCII code for zero.

thomastaylor312 reviewed (approved):
Tested this and it looks good to go
